### ISSUE TITLE
Add row(s) to scenes tab for a better overview of open scenes

### ISSUE THREAD

**Issue by peterhoglund** [godotengine/godot-proposals#9954]
### Describe the project you are working on

Editor feature

### Describe the problem or limitation you are having in your project

Quite often you tend to have many scenes open at the same time, especially with more complex features where your structure is broken down in many scenes or when dealing with resources. The open scenes tab bar quickly fills up and scenes are hidden behind the arrow buttons to the right. This makes it hard to get a good overview of the open scenes and navigating between them is slow. This is especially true for users on laptops.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Expand the tab bar with multiple rows to show all open scenes. The number of rows can be defined in a setting but defaulting to 2 rows seems reasonable (the user should of course still be able to choose 1 row). If the tabs overflow the arrow buttons are still there like today. 

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

This is what it looks like today:
![image](https://github.com/godotengine/godot-proposals/assets/522470/9fdc5699-803a-46e4-ac7d-c3f2c72c0c7c)

Here is a mockup of what multiple rows would look like:
![image](https://github.com/godotengine/godot-proposals/assets/522470/1bf4d9fe-1604-42da-8b09-9a7a08404e1b)

### If this enhancement will not be used often, can it be worked around with a few lines of script?

-

### Is there a reason why this should be core and not an add-on in the asset library?

Editor core feature

**Comment by FabriceCastel** [comment]
If this gets implemented, that behaviour should be togglable in the editor settings. A neat way to expose this with the right amount of flexibility IMO would be as a `max_scene_tab_rows` (defaults to `1`) to give you more control over how much it should be allowed to grow.

**Comment by peterhoglund** [comment]
@Calinou Not really, I think. If you spam the middle mouse button on the first row the mouse pointer will still be in the first row when the change happens. And if you spam it on the second row, this will only be an issue if you press the first item, and as the row disappears you might middle mouse click some of the first buttons in the toolbar, which I don't think will do anything. so even though it might look a bit weird that the row disappears, it will not be a destructive action. To me personally, having a better overview of my scenes feels more important than a small annoyance like that.

**Comment by peterhoglund** [comment]
I can see two alternative solutions to this for a similar result.
1. Compress the tab size when out of space, similar to how many browsers do it (not optimal since you usually have many scenes starting with the same word so you won't be able to differentiate them)
2. Be able to scroll with the scroll wheel when hovering over the tab bar. It doesn't solve the improved overview, but at least it is faster than clicking the small arrow button.

**Comment by Mickeon** [comment]
> Be able to scroll with the scroll wheel when hovering over the tab bar.

All tab bars for benefit from this, actually.

Also this entire proposal would complement https://github.com/godotengine/godot-proposals/issues/4081 nicely, due of tab rows being implemented as an actual feature.

**Comment by KoBeWi** [comment]
Another option is hiding extra tabs in a dropdown menu. Selecting a hidden tab could make it replace the currently last one.

**Comment by AdriaandeJongh** [comment]
Horizontal scroll and / or a dropdown for additional opened scenes that don't fit are both much better UX than the arrows, so it's love to see this built.

One suggestion for the dropdown: also list the visible tabs there. It's easy to miss a visible tab if there's many, and not listing them risks the user thinking it's not opened.

**Comment by MotherLeopard** [comment]
I would love to have the option to have several rows of tabs for my open scene. I find myself irritated every time I have to switch between my open scenes, because so many of them becomes hidden. Since pretty much everything is a Scene in Godot, I have a lot of them I want to jump around between easily while working, and the very small arrows that I have to use to go to the hidden tabs are too cumbersome to use imo. 

So, I'd love to be able to use multiple rows. I'd also love to have a dropdown menu of all currently open scenes. 

All of this would really help me in combating my shoulder pain that I easily get from working at the computer in Godot. It would reduce my irritation (which makes me tense up) and also reduce the amount of clicks to perform an action.

**Comment by FabriceCastel** [comment]
@MotherLeopard an alternative to alleviate your shoulder pain might be to start using shortcuts to navigate between scenes; `ctrl+shift+O`, so you don't have to use the mouse at all to switch scenes - at least until something like this gets implemented.

**Comment by arkology** [comment]
> in a dropdown menu

I have a [branch](https://github.com/arkology/godot/tree/scenes-list) with something like this (UPD: related to https://github.com/godotengine/godot-proposals/issues/9642). Functionality-wise it works. 
![Image](https://github.com/user-attachments/assets/492df9af-0e2c-4f96-86aa-be756f214d1d)

But on UI side I have problems with handling dropdown menu button position (you could see it even on above screenshot, three-dot button is misaligned). This problems comes from assumption that dropdown menu button should be hidden if all scene tabs are visible (and I'm trying to make it work same way as `scene_tab_add` button). But at the same time Visual Studio, for example, always shows such button. 

So I need an advise, should it be always visible or only when not all scene tabs are visible. If it is ok for this button being always visible, I could open a PR even tomorrow. @KoBeWi what do you think?

**Comment by KoBeWi** [comment]
I think it's fine to always show it.

**Comment by Mickeon** [comment]
While this is nice, I'm not sure https://github.com/godotengine/godot/pull/108079 directly solves this proposal. Tabs do not expand into rows, nor is the feature possible in the engine currently.
